agimus_controller/agimus_controller/ocp_base_croco.py

```diff
@@ -93,15 +95,63 @@ def create_terminal_model(self) -> crocoddyl.ActionModelAbstract:
         """Create the terminal model."""
         pass
 
-    def set_reference_weighted_trajectory(
-        self, reference_weighted_trajectory: list[WeightedTrajectoryPoint]
-    ):
+    def set_reference_weighted_trajectory(self):
         """Set the reference trajectory for the OCP."""
-        if self._ocp_params.use_debug_data:
-            reference_trajectory_points = [
-                el.point for el in reference_weighted_trajectory
+        pass
+
+    def get_nb_iterations(self):
+        """Get max number of solver iterations and qp iterations."""
+        return self._ocp_params.solver_iters, self._ocp_params.qp_iters
+
+    def set_nb_iterations(self, nb_iter, qp_iters):
+        """Set max number of solver iterations and qp iterations."""
+        self._solver.max_qp_iters = qp_iters
+        self._ocp_params.qp_iters = qp_iters
+        self._ocp_params.solver_iters = nb_iter
+
+    def fill_debug_data(self, res, ocp_results):
+        """Fill attributes of dataclass OCPDebugData."""
+        self._debug_data.problem_solved = res
+        self._debug_data.result = ocp_results
+        self._debug_data.kkt_norm = self._solver.KKT
+        self._debug_data.nb_iter = int(self._solver.iter)
+        self._debug_data.nb_qp_iter = int(self._solver.qp_iters)
+        # TODO the keys used for residuals and references should be parameters
+        if len(self._debug_data.residuals) == 0:
```

Comment:
At initialization time, I would store a list of tuple (name, list_of_residual_model_or_data) as an attribute of the class. This would make the loop faster at runtime and easier to read.